sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java

```diff
@@ -133,6 +156,63 @@ public AzureEnvironment(Map<String, String> endpoints) {
             put("azureApplicationInsightsResourceId", "N/A");
         }});
 
+    /**
+     * Gets AzureEnvironment from Azure Resource Manager endpoint.
+     *
+     * If multiple Azure Resource Manager metadata is available, it uses the first one.
+     * Note that the Microsoft Graph endpoint currently is not available from metadata.
+     *
+     * @param endpoint the URL of Azure Resource Manager endpoint.
+     * @param httpPipeline the HTTP pipeline for requests.
+     * @return the AzureEnvironment
+     * @throws HttpResponseException thrown if the request is rejected by server, or failed to parse metadata.
+     * @throws IllegalArgumentException thrown if Azure Resource Manager endpoint is invalid.
+     * @throws NullPointerException thrown if Azure Resource Manager endpoint is null.
+     * @throws URISyntaxException thrown if Azure Resource Manager endpoint violates RFC 2396.
+     */
+    public static AzureEnvironment fromAzureResourceManagerEndpoint(String endpoint, HttpPipeline httpPipeline)
+        throws URISyntaxException {
+
+        // verify endpoint
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        URI endpointUri = new URI(endpoint);
+        endpoint = endpointUri.toString();
+        if (!endpoint.endsWith("/")) {
+            endpoint += "/";
+        }
+        String metadataEndpoint = String.format("%smetadata/endpoints?api-version=2019-10-01", endpoint);
```

Comment:
Having a hardcoded API version should be avoided. This makes the argument for having a separate client to fetch metadata stronger.

Reply:
Partly agree. This `metadata/endpoints` API appears not well documented. If we do not know the `api-version`, user probably will less likely know it.